tests/testthat/test-archive.R

```diff
@@ -92,7 +92,7 @@ test_that("epi_archives are correctly instantiated with a variety of data types"
   expect_equal(key(ea1$DT), c("geo_value", "time_value", "version"))
 
   ea2 <- as_epi_archive(df, other_keys = "value", compactify = FALSE)
-  expect_equal(key(ea2$DT), c("geo_value", "time_value", "value", "version"))
+  expect_equal(key(ea2$DT), c("geo_value", "value", "time_value", "version"))
```

Comment:
where did this order change come from? It does seem like an odd order; why isn't it geo_value, time_value, version, value?

Reply:
This probably comes from us moving from geo_value, time_value, other_keys[, version] canonical ordering to geo_value, other_keys, time_value[, version].  It looks pretty weird here probably moreso due to using the column name "value" rather than "age_group" or something like that.

Reply:
Sorry, actually we're not canonically ordering archive cols themselves in `as_epi_archive`, but are at least using the new canonically ordering of the vars in the key.  Change presumably comes from
```r
key_vars <- c("geo_value", other_keys, "time_value", "version")
```
line, but it's mixed up in some changes addressing data.table footguns so it's not quite obvious.